### PR TITLE
Add wrapper for SemiBin

### PR DESCRIPTION
**Pull request by bebatut**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)
---

Missing
- Test data for the reference database 
- Fix for multiple taxonomy inputs (waiting for new conda release)

### CODE REVIEW

File: tools/semibin/semibin.xml [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+    --epoches $training.epoches
+    --batch-size $training.batch_size
+    --max-node $bin.max_node
+    --max-edges $bin.max_edges
+    --minfasta-kbs $bin.minfasta_kbs
+    $bin.no_recluster
+    --threads \${GALAXY_SLOTS:-1}
+    --processes \${GALAXY_SLOTS:-1}
+&& 
+echo "output" &&
```
Comment (bgruening) [inline tools/semibin/semibin.xml:59]: remove debug messages

File: tools/semibin/semibin.xml [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+                </element>
+                <element name="2" ftype="csv">
+                    <assert_contents>
+                        <has_text text="g1k_0_1"/>
+                        <has_text text="g2k_7_2"/>
+                    </assert_contents>
+                </element>
+            </output_collection>
+        </test>
+        <!--<test expect_num_outputs="4">
```
Comment (bgruening) [inline tools/semibin/semibin.xml:292]: What about those tests? Please add a reason those are not working for later.

File: tools/semibin/train.xml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                <param name="method" value="min-len"/>
+                <param name="min_len" value="2500" />
+            </conditional>
+            <param name="orf_finder" value="prodigal"/>
+            <param name="random-seed" value="0"/>
+            <param name="epoches" value="1"/>
+            <param name="batch_size" value="2048"/>
+            <output name="model" ftype="h5">
+                <assert_contents>
+                    <has_size value="3119500" delta="100" />
```
Comment (bgruening) [inline tools/semibin/train.xml:109]: hdf has some very nice checks, eg. for layers etc ... those are better tests imho

File: tools/semibin/generate_cannot_links.xml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            <param name="ml_threshold" value=""/>
+            <output name="cannot" ftype="txt">
+                <assert_contents>
+                    <has_text text="g1k_0,g4k_0"/>
+                    <has_text text="g2k_3,g4k_5"/>
+                    <has_text text="g4k_6,g4k_5"/>
+                </assert_contents>
+            </output>
+        </test>
+        <!--<test expect_num_outputs="1">
```
[the remark this answers is not in the document]
Comment (bebatut) [reply tools/semibin/generate_cannot_links.xml:99]: This test was with the r multiple taxonomy inputs. I was waiting for new conda release that fixed a bug

File: tools/semibin/semibin.xml (hunk 1 above)
Comment (bebatut) [reply tools/semibin/semibin.xml:59]: Done

File: tools/semibin/semibin.xml (hunk 2 above)
Comment (bebatut) [reply tools/semibin/semibin.xml:292]: This test was with the r multiple taxonomy inputs. I was waiting for new conda release that fixed a bug

File: data_managers/data_manager_semibin/data_manager/data_manager_semibin.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        table_name,
+        dict(
+            dbkey='gtdb',
+            value='%s' % (date.today().strftime("%d%m%Y")),
+            name="GTDB reference genome generated by MMseqs2 used in SemiBin",
+            path=str(db_dp)))
+
+
+if __name__ == "__main__":
+    print("Starting...")
```
[the remark this answers is not in the document]
Comment (bebatut) [reply data_managers/data_manager_semibin/data_manager/data_manager_semibin.py:112]: Removed

File: tools/semibin/train.xml (hunk 3 above)
Comment (bebatut) [reply tools/semibin/train.xml:109]: I do not know what are the layers or keys in there. I tried to find that info but no success

**Comment by bebatut** [comment]
It is failing on the data manager: `Timed out after 900.25 seconds waiting on tool test run.`

**Comment by bebatut** [comment]
I fixed the `Time out` but I do not understand now why the data manager test is failing...

**Comment by bebatut** [comment]
Yeah the data manager test passes now. 
About the file size, I worked with the tool developers to get the test data and we can not really reduce it further

**Review by bgruening — APPROVED** [review]

**Pull request merged**